core/state/intra_block_state.go

```diff
@@ -825,3 +826,46 @@ func (sdb *IntraBlockState) AddressInAccessList(addr libcommon.Address) bool {
 func (sdb *IntraBlockState) SlotInAccessList(addr libcommon.Address, slot libcommon.Hash) (addressPresent bool, slotPresent bool) {
 	return sdb.accessList.Contains(addr, slot)
 }
+
+// ValidateKnownAccounts validates the knownAccounts passed in the options parameter in the conditional transaction (EIP-4337)
+func (sdb *IntraBlockState) ValidateKnownAccounts(knownAccounts types.KnownAccounts) error {
```

Comment:
Could you please recommend any other place for this? The reason I added it here was because the storage slots were available at this location.

About the comment, it is not in the EIP, but in the spec released by the EF researchers here - https://notes.ethereum.org/@yoav/SkaX2lS9j.

Reply:
We'd like to have a new top-level `bor` folder/package for this with intention to centralize the other bor-specific code into it in the future.
If you make a new file like `bor/transaction_conditional.go` and move
ValidateKnownAccounts and ValidateTransactionConditions functions there, would it make sense to you?